c10/core/DispatchKey.h

```diff
@@ -353,8 +353,8 @@ enum class DispatchKey : uint16_t {
   AutocastCPU,
   AutocastXPU,
   AutocastHPU,
-  // Naughtily, AutocastCUDA is also being used for XLA.  In the terminal state,
-  // it probably should get its own Autocast key
+  AutocastXLA, 
+  // AutocastXLA is only being used for TPUs. XLA GPUs continue to use AutocastCUDA.
```

Comment:
@cowanmeg can you describe how this works a bit more - what's the UX here? Is the user expected to use `torch.cuda.autocast()` when using XLA with gpu's, and torch.xla.autocast()` when using tpu's?

Reply:
Correct. Updated the summary for clarity.